util/jsonrpc-types/src/uint64.rs

```diff
@@ -0,0 +1,136 @@
+use ckb_types::prelude::{Pack, Unpack};
+use ckb_types::{core, packed};
+use serde::de::{Error, Visitor};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::fmt;
+
+#[derive(Copy, Clone, Default, PartialEq, Eq, Hash, Debug)]
+pub struct Uint64(u64);
+
+impl Uint64 {
+    pub fn value(self) -> u64 {
+        self.0
+    }
+}
+
+impl From<core::Capacity> for Uint64 {
+    fn from(value: core::Capacity) -> Self {
+        Uint64(value.as_u64())
+    }
+}
+
+impl From<Uint64> for core::Capacity {
+    fn from(value: Uint64) -> Self {
+        core::Capacity::shannons(value.value())
+    }
+}
+
+impl From<u64> for Uint64 {
+    fn from(value: u64) -> Self {
+        Uint64(value)
+    }
+}
+
+impl From<Uint64> for u64 {
+    fn from(value: Uint64) -> Self {
+        value.value()
+    }
+}
+
+impl Pack<packed::Uint64> for Uint64 {
+    fn pack(&self) -> packed::Uint64 {
+        self.value().pack()
+    }
+}
+
+impl Unpack<Uint64> for packed::Uint64 {
+    fn unpack(&self) -> Uint64 {
+        Uint64(Unpack::<u64>::unpack(self))
+    }
+}
+
+impl Serialize for Uint64 {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        serializer.serialize_str(&format!("0x{:x}", self.0))
+    }
+}
+
+impl<'a> Deserialize<'a> for Uint64 {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'a>,
+    {
+        deserializer.deserialize_any(Uint64Visitor)
+    }
+}
+
+struct Uint64Visitor;
+
+impl<'a> Visitor<'a> for Uint64Visitor {
+    type Value = Uint64;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a hex-encoded or decimal uint64")
+    }
+
+    fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        match value {
+            _ if value.starts_with("0x") => u64::from_str_radix(&value[2..], 16)
```

Comment:
I'm not sure about this:

```rust
fn main () {
    let x1 = u32::from_str_radix("0", 16).unwrap();
    let x2 = u32::from_str_radix("00", 16).unwrap();
    let x3 = u32::from_str_radix("000", 16).unwrap();
    let x4 = u32::from_str_radix("0000", 16).unwrap();
    assert_eq!(x1, x2);
    assert_eq!(x1, x3);
    assert_eq!(x1, x4);
}
```

Should we remove this ambiguity?

Reply:
I have no strong opinions for it. 

@ashchan Any suggestions?

Reply:
The current/present situation:
- Return numbers in heximal format(**without** leading zeros)
- U32/U64: Accept numbers in heximal format(**with** any number leading zeros)
- U256:  Accept numbers in heximal format(**without** leading zeros, expected `0x0`)

Reply:
Agree!
So, I should make a more strict limitation on json-numbers: no leading zeros, no mather inputs or outputs, expect `0x0`.